orbit/pkg/useraction/mdm_migration_darwin.go

```diff
@@ -0,0 +1,287 @@
+	secondaryBtnExitCode = 2
+	infoBtnExitCode      = 3
+	timeoutExitCode      = 4
+	userQuitExitCode     = 10
+	unknownExitCode      = 99
+)
+
+var mdmMigrationTemplate = template.Must(template.New("mdmMigrationTemplate").Parse(`
+## Migrate to Fleet
+
+To begin, click "Start." Your default browser will open your My Device page.
+
+{{ if .Aggressive }}You {{ else }} Once you start, you {{ end -}} will see this dialog every 15 minutes until you click "Turn on MDM" and complete the instructions.
+
+\\![Image showing the Fleet UI](http://localhost:64793/mdm-migration-screenshot-768x180@2x.jpg)
+
+Unsure? Contact {{ .OrgInfo.OrgName }} IT [here]({{ .OrgInfo.ContactURL }}).
+`))
+
+var errorTemplate = template.Must(template.New("").Parse(`
+### Something's gone wrong.
+
+Please contact your IT admin [here]({{ .ContactURL }}).
+`))
+
+// baseDialog implements the basic building blocks to render dialogs using
+// swiftDialog.
+type baseDialog struct {
+	path        string
+	interruptCh chan struct{}
+}
+
+func newBaseDialog(path string) *baseDialog {
+	return &baseDialog{path: path, interruptCh: make(chan struct{})}
+}
+
+func (b *baseDialog) CanRun() bool {
+	if _, err := os.Stat(b.path); err != nil {
+		return false
+	}
+
+	return true
+}
+
+// Exit sends the interrupt signal to try and stop the current swiftDialog
+// instance.
+func (b *baseDialog) Exit() {
+	b.interruptCh <- struct{}{}
+	log.Info().Msg("dialog exit message sent")
+}
+
+// render is a general-purpose render method that receives the flags used to
+// display swiftDialog, and starts an asyncronous routine to display the dialog
+// without blocking.
+//
+// The first returned channel sends the exit code returned by swiftDialog, and
+// the second channel is used to send errors.
+func (b *baseDialog) render(flags ...string) (chan swiftDialogExitCode, chan error) {
+	exitCodeChan := make(chan swiftDialogExitCode)
+	errChan := make(chan error)
+	go func() {
+		cmd := exec.Command(b.path, flags...) //nolint:gosec
+		done := make(chan error)
+
+		if err := cmd.Start(); err != nil {
+			errChan <- err
+			return
+		}
+
+		go func() { done <- cmd.Wait() }()
+
+		select {
+		case <-b.interruptCh:
+			if err := cmd.Process.Signal(os.Interrupt); err != nil {
+				log.Error().Err(err).Msg("sending interrupt signal to swiftDialog process")
+				if err := cmd.Process.Kill(); err != nil {
+					log.Error().Err(err).Msg("killing swiftDialog process")
+				}
+
+			}
+		case err := <-done:
+			if err != nil {
+				// non-zero exit codes
+				if exitError, ok := err.(*exec.ExitError); ok {
+					ec := exitError.ExitCode()
+					switch ec {
+					case errorExitCode:
+						exitCodeChan <- errorExitCode
+					case secondaryBtnExitCode, infoBtnExitCode, timeoutExitCode:
+						exitCodeChan <- swiftDialogExitCode(ec)
+					default:
+						errChan <- fmt.Errorf("unknown exit code showing dialog: %w", exitError)
+					}
+				}
+
+				errChan <- fmt.Errorf("running swiftDialog: %w", err)
```

Comment:
This should be in an `else` clause of the `if` above, because as it stands it could send two errors in the `errChan`, one in the `default` case of the `switch`, and one here. Callers will only look to receive once, so the next send would block forever and leak that goroutine.